# PHT EB Candidates - Comparison with TESS EB

[TESS EB Catalog](http://tessebs.villanova.edu/) uses PHT #eclipsingbinary tagging as one of the input sources.

In [1]:
from importlib import reload

from IPython.display import display, HTML, clear_output

import numpy as np

import pandas as pd
from pandas.io.clipboard import clipboard_set

import matplotlib.pyplot as plt

import catalog_stats
import dashboard_utils
from dashboard_utils import CAT_COLS_COMMON

display(HTML("<style>.container { width:99% !important; }</style>"))

- To compare with the result from  the published [TESS EB catalog](https://ui.adsabs.harvard.edu/abs/2022ApJS..258...16P/abstract), covering sectors 1 - 26, we use a subset of PHT EB Catalog that has comparable coverage.
- Furthermore, we only consider the subset that has decent accuracy (`eb_score >= 3`)

In [2]:
reload(dashboard_utils)

df_catplus = dashboard_utils.join_pht_eb_candidate_catalog_with(dashboard_utils.get_catalog("pht_eb"), ["tesseb"])

# Also exclude those observed both in year 1 and year 3, to be on the conservative side
df_catplus = df_catplus[df_catplus["max_sector"] <= 26]

print("Num of TICs:", len(df_catplus))

min_eb_score = 3
res, report, styler = catalog_stats.estimate_num_ebs_not_in_catalog(df_catplus, min_eb_score=min_eb_score);
display(report)

df_subset = df_catplus[df_catplus["eb_score_group"] == "03+"]
display(HTML(f"Num of TICs with decent accuracy: {len(df_subset)}"))

Num of TICs: 7874


count                            
               tic_id                            
is_eb_catalog       T    F     - Totals   T/(T+F)
eb_score_group                                   
03+              1514   88   665   2267  0.945069
02-              1205  782  3620   5607  0.606442
Totals           2719  870  4285   7874  0.757593

### Those in PHT EB but not in TESS EB

- are they false positives in PHT EB, or could it be that TESS EB miss those targets?
  - TESS EB does miss a noticeable portion of them:
  - 246 of them have been classified as EB in SIMBAD, VSX or ASAS-SN 
- If TESS EB miss those targets, are there some (systematic) reasons?

In [3]:
report = df_subset.pivot_table(
    index=["TESSEB_Is_In"],
    columns="is_eb_catalog",
    values=["tic_id"],
    aggfunc=["count"],
    margins=True,
    margins_name="Totals",
)

report

count                 
              tic_id                 
is_eb_catalog      -   F     T Totals
TESSEB_Is_In                         
F                136  33   246    415
T                529  55  1268   1852
Totals           665  88  1514   2267

- 415 out of 2267 TICs are not in TESS EB. Out of 415
    - 246 of them are already classified as EB in SIMBAD , VSX or ASAS-SN.
    - 166 (136 + 33) of them remain uncertain, and require further triage

### Other considerations

TODO:
- TESS EB false positives due to NEB
- TESS EB eclipse duration accuracy (the TESS EB paper had some discussion)


### Appendix: Sample TICs

#### Sample TICs that TESS EB missed already classified as EB in SIMBAD, VSX or ASAS-SN

- they are likely real miss in TESS EB, given the existing classification
- possibly worth some investigation on if there are some systematic issues leading to their misses.

In [9]:
df = df_subset
df = df[(df["TESSEB_Is_In"] != "T")]
df = df[(df["is_eb_catalog"] == "T")]

start, end = 0, 15

# sample top TICs
df = df.sort_values(["eb_score", "tic_id"], ascending=[False, True])
display(dashboard_utils.style(df[["best_subject_img_id", ] + CAT_COLS_COMMON][start:end], show_thumbnail=True))

# sample average-ish TICs (eb_score is 3)
df = df.sort_values(["eb_score", "tic_id"], ascending=[True, True])
display(dashboard_utils.style(df[["best_subject_img_id", ] + CAT_COLS_COMMON][start:end], show_thumbnail=True))

- no obvious pattern found yet, e.g., their VSX classifications (most of them have one) follow general trend (EA followed by EW)

In [16]:
df_vsx = df[~pd.isna(df["VSX_Type"])]
display(HTML(f"Top 5 VSX Classification for those not in TESS EB ( {len(df_vsx)} out of {len(df)} TICs with VSX classification))"))
df_vsx.groupby("VSX_Type").agg(
    num_tics=("tic_id", "count"),
).sort_values("num_tics", ascending=False).head(5)


num_tics
VSX_Type          
EA             101
EW              44
EC               9
EB               9
EA/RS            7

#### Sample TICs that TESS EB missed, but not in SIMBAD, VSX or ASAS-SN

- they require more vetting, as they tend to have a mix of false positives and genuine EBs.

In [5]:
df = df_subset
df = df[(df["TESSEB_Is_In"] != "T")]
df = df[(df["is_eb_catalog"] != "T")]


start, end = 0, 15

# sample top TICs
df = df.sort_values(["eb_score", "tic_id"], ascending=[False, True])
display(dashboard_utils.style(df[["best_subject_img_id", ] + CAT_COLS_COMMON][start:end], show_thumbnail=True))

# sample average-ish TICs (eb_score is 3)
df = df.sort_values(["eb_score", "tic_id"], ascending=[True, True])
display(dashboard_utils.style(df[["best_subject_img_id", ] + CAT_COLS_COMMON][start:end], show_thumbnail=True))
